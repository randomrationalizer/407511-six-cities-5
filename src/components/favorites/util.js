export const getCitiesFromOffers = (offers) => {
  const cities = offers.slice().map((offer) => offer.city.name).sort();
  return [...new Set(cities)];
};
