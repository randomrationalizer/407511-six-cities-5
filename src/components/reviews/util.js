const REVIEW_MIN_LENGTH = 50;
const REVIEW_MAX_LENGTH = 300;
const RATING_MIN = 1;
const RATING_MAX = 5;

export const checkCommentValidity = (comment) => {
  return comment.length >= REVIEW_MIN_LENGTH && comment.length <= REVIEW_MAX_LENGTH ? true : false;
};

export const checkRatingValidity = (rating) => {
  return rating >= RATING_MIN && rating <= RATING_MAX ? true : false;
};
