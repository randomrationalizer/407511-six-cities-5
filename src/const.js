export const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const DEFAULT_CITY = `Paris`;

export const OfferType = {
  MAIN: `cities`,
  NEARBY: `near-places`,
  FAVORITES: `favorites`
};

export const OfferPageType = {
  CARD: `CARD`,
  DETAILS: `DETAILS`
};

export const CityLinkType = {
  MAIN_PAGE: `MAIN_PAGE`,
  INNER_PAGE: `INNER_PAGE`
};

export const MapType = {
  MAIN: `cities`,
  PROPERTY: `property`
};

export const SortType = {
  DEFAULT: `DEFAULT`,
  PRICE_UP: `PRICE_UP`,
  PRICE_DOWN: `PRICE_DOWN`,
  RATING_DOWN: `RATING_DOWN`
};

export const KeyCode = {
  ENTER: 13,
  SPACE: 32
};

export const HttpCode = {
  OK: 200,
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppRoute = {
  MAIN: `/`,
  OFFERS: `/hotels`,
  FAVORITES: `/favorite`,
  LOGIN: `/login`,
  NOT_FOUND: `/404`
};

export const APIRoute = {
  OFFERS: `/hotels`,
  NEARBY: `nearby`,
  FAVORITES: `/favorite`,
  REVIEWS: `/comments`,
  LOGIN: `/login`
};

export const ErrorMessage = {
  BAD_REQUEST: `Invalid data format`,
  NETWORK_ERROR: `Network unavailable`
};
