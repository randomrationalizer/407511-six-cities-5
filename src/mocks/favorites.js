export const favorites = [`1`, `3`, `4`];
