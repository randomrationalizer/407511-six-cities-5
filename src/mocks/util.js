import configureStore from "redux-mock-store";

export const noop = () => {};

export const getMockStore = (store) => {
  const storeConfig = configureStore();
  return storeConfig(store);
};

export const nope = () => {
  return Promise.resolve({});
};
