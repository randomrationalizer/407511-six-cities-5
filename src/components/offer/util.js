import dayjs from "dayjs";
import {OfferType} from "../../const";


export const capitalize = (str) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

// Возвращает дату в виде строки в формате: 2019-04-24
export const formatDate = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

// Возвращает дату в виде строки в формате: January 2020
export const humanizeDate = (date) => {
  return dayjs(date).format(`MMMM YYYY`);
};

// Сравнение для сортировки отзывов в хронологическом порядке
export const sortByDate = (a, b) => {
  return dayjs(a.date) - dayjs(b.date);
};

export const getDescriptionSentences = (text) => {
  let description = text.split(`. `);
  description[description.length - 1] = description[description.length - 1].replace(`.`, ``);

  return description;
};

export const isFavoritesCard = (type) => {
  return type === OfferType.FAVORITES;
};

export const isMainPageCard = (type) => {
  return type === OfferType.MAIN;
};
