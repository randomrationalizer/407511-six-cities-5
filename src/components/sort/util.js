import {SortType} from "../../const";

export const sortTypeToTitle = {
  [SortType.DEFAULT]: `Popular`,
  [SortType.PRICE_UP]: `Price: low to high`,
  [SortType.PRICE_DOWN]: `Price: high to low`,
  [SortType.RATING_DOWN]: `Top rated first`
};
