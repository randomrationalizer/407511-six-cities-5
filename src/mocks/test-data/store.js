import configureStore from "redux-mock-store";
import {NameSpace} from "../../store/root-reducer";
import mockOffers from "./offers";
import mockReviews from "./reviews";
import mockNearbyOffers from "./nearby-offers";
import mockCurrentOffer from "./current-offer";
import mockUserInfo from "./user-info";
import mockFavorites from "./favorites";
import mockCities from "./cities";
import {SortType, AuthorizationStatus, DEFAULT_CITY} from "../../const";


export default {
  [NameSpace.APP_DATA]: {
    currentCity: DEFAULT_CITY,
    allOffers: mockOffers,
    cities: mockCities,
    currentSort: SortType.DEFAULT
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
    isAuthRequestComplete: true,
    userInfo: mockUserInfo,
    favorites: mockFavorites
  },
  [NameSpace.CURRENT_OFFER]: {
    offer: mockCurrentOffer,
    reviews: mockReviews,
    nearbyOffers: mockNearbyOffers
  },
  [NameSpace.LOAD_STATUS]: {
    isOffersLoaded: true,
    isFavoritesLoaded: true,
    isCurrentOfferLoaded: true,
    errorMessage: null,
    isLoadFinished: true
  }
};
