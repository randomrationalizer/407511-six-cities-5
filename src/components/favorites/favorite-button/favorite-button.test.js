import React from "react";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import FavoriteButton from "./favorite-button";
import {OfferPageType} from "../../../const";
import {getMockStore, noop} from "../../../mocks/util";
import store from "../../../mocks/test-data/store";


const mockStore = getMockStore(store);

jest.mock(`react-router`, () => (Object.assign({},
    jest.requireActual(`react-router-dom`),
    {
      useHistory: () => {},
      useLocation: () => {}
    }
)));

describe(`Should FavoriteButton component renders correctly`, () => {
  describe(`with active state`, () => {
    it(`for property page`, () => {
      const tree = renderer
        .create(
            <Provider store={mockStore}>
              <FavoriteButton
                id={1}
                isActive={true}
                pageType={OfferPageType.DETAILS}
                showErrorMessage={noop}
              />
            </Provider>
        ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`for offer card`, () => {
      const tree = renderer
        .create(
            <Provider store={mockStore}>
              <FavoriteButton
                id={1}
                isActive={true}
                pageType={OfferPageType.CARD}
                showErrorMessage={noop}
              />
            </Provider>
        ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`with inactive state`, () => {
    it(`for property page`, () => {
      const tree = renderer
      .create(
          <Provider store={mockStore}>
            <FavoriteButton
              id={1}
              isActive={false}
              pageType={OfferPageType.DETAILS}
              showErrorMessage={noop}
            />
          </Provider>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`for offer card`, () => {
      const tree = renderer
      .create(
          <Provider store={mockStore}>
            <FavoriteButton
              id={1}
              isActive={false}
              pageType={OfferPageType.CARD}
              showErrorMessage={noop}
            />
          </Provider>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
