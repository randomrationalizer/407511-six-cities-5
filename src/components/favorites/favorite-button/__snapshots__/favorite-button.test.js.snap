// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteButton component renders correctly with active state for offer card 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`Should FavoriteButton component renders correctly with active state for property page 1`] = `
<button
  className="property__bookmark-button property__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height={33}
    width={31}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`Should FavoriteButton component renders correctly with inactive state for offer card 1`] = `
<button
  className="place-card__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`Should FavoriteButton component renders correctly with inactive state for property page 1`] = `
<button
  className="property__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height={33}
    width={31}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
